server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/tree/logic/SectionNodeProcessorHelper.java

```diff
@@ -237,22 +235,40 @@ private List<SyntaxError> processDefinition(
         }
         definitionNodes.removeFirst();
       } else {
-        Node parentForNextLevel =
-            Optional.ofNullable(lastVariableOnTheLevel)
-                .filter(
-                    it ->
-                        it.getVariableType() == VariableType.GROUP_ITEM
-                            || it.getVariableType() == VariableType.MULTI_TABLE_DATA_NAME
-                            || definitionNode.getLevel() == LEVEL_88)
-                .map(Node.class::cast)
-                .orElse(rootNode);
         errors.addAll(
-            processDefinition(parentForNextLevel, definitionNode.getLevel(), definitionNodes));
+            processDefinition(
+                getParentForNextLevel(rootNode, lastVariableOnTheLevel, definitionNode),
+                definitionNode.getLevel(),
+                definitionNodes));
       }
     }
     return errors;
   }
 
+  private boolean isRootLevel(int level) {
+    return ImmutableSet.of(LEVEL_66, LEVEL_77, LEVEL_88, LEVEL_01).contains(level);
```

Comment:
Not sure, but I guess 88 can be under a group item

Reply:
That makes sense. There is a check for it before the new logic. I'll remove it from the list here.